proof-systems/src/darlin/benches/batch_verification.rs

```diff
@@ -0,0 +1,112 @@
+use algebra::{AffineCurve, ToConstraintField};
+use poly_commit::{
+    PolynomialCommitment,
+    ipa_pc::InnerProductArgPC
+};
+use proof_systems::darlin::{
+    tests::{
+        get_keys,
+        final_darlin::generate_test_data as generate_final_darlin_test_data
+    },
+    proof_aggregator::batch_verify_proofs,
+};
+use digest::Digest;
+use criterion::*;
+use rand::{thread_rng, SeedableRng};
+use blake2::Blake2s;
+use proof_systems::darlin::pcd::GeneralPCD;
+use rand_xorshift::XorShiftRng;
+
+fn bench_batch_verification<G1: AffineCurve, G2: AffineCurve, D: Digest>(
+    c: &mut Criterion,
+    bench_name: &str,
+    segment_size: usize,
+    max_proofs: Vec<usize>,
+)
+    where
+        G1: AffineCurve<BaseField = <G2 as AffineCurve>::ScalarField> + ToConstraintField<<G2 as AffineCurve>::ScalarField>,
+        G2: AffineCurve<BaseField = <G1 as AffineCurve>::ScalarField> + ToConstraintField<<G1 as AffineCurve>::ScalarField>,
+{
+    let rng = &mut XorShiftRng::seed_from_u64(1234567890u64);
+    let mut group = c.benchmark_group(bench_name);
+    let num_constraints = 1 << 19;
+
+    //Generate DLOG keys
+    let params_g1 = InnerProductArgPC::<G1, D>::setup(segment_size - 1).unwrap();
+    let params_g2 = InnerProductArgPC::<G2, D>::setup(segment_size - 1).unwrap();
+
+    let (
+        _, verifier_key_g1,
+        _, verifier_key_g2
+    ) = get_keys::<_, _, D>(&params_g1, &params_g2);
+
+    let (final_darlin_pcd, index_vk) = generate_final_darlin_test_data::<G1, G2, D, _>(
+        num_constraints,
+        segment_size,
+        &params_g1,
+        &params_g2,
+        1,
+        rng
+    );
```

Comment:
Same as in accumulate_verify benches: why only a single instance of PCD?

Reply:
As above